sklearn/decomposition/nmf.py

```diff
@@ -1106,7 +1106,6 @@ def transform(self, X):
             nls_max_iter=self.nls_max_iter, sparseness=self.sparseness,
             beta=self.beta, eta=self.eta)
 
-        self.n_iter_ = n_iter_
```

Comment:
Very awkwardly, there is an `Attributes` section in the docstring above (and similar ones in the docs for `fit` and `fit_transform`!). Please remove it. I suppose this change should also be documented ("Fixed bug where `NMF`'s `n_iter_` attribute was set by calls to `transform`"), though it's a very strange feature.